pkgs/development/compilers/idris2/default.nix

```diff
@@ -0,0 +1,41 @@
+{ stdenv, fetchFromGitHub, makeWrapper
+, clang, chez 
+}:
+
+# Uses scheme to bootstrap the build of idris2
+stdenv.mkDerivation {
+  name = "idris2";
+  version = "0.2.0-840e020";
+
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ clang chez ];
+
+  prePatch = ''
+    patchShebangs --build tests
+  '';
+
+  makeFlags = "PREFIX=$(out)";
+
+  # The name of the main executable of pkgs.chez is `scheme`
+  buildFlags = "bootstrap SCHEME=scheme";
+
+  # idris2 needs to find scheme at runtime to compile
+  postInstall = ''
+    wrapProgram "$out/bin/idris2" --prefix PATH : "${chez}/bin"
+  '';
+
+  src = fetchFromGitHub {
+    owner = "idris-lang";
+    repo = "Idris2";
+    rev = "840e020d8ccc332135e86f855ad78053ca15d603";
+    sha256 = "1l6pdjiglwd13pf56xwzbjzyyxgz48ypfggjgsgqk2w57rmbfy90";
+  };
```

Comment:
It is quite common to put the src after name and version, but not a written rule.

Reply:
Done. Thanks